### PR TITLE
FC: jailer failed when importing new flag "--config-file"

### PR DESCRIPTION
**Pull request by Pennyzct**
# Description of problem
When we used jailer to launch firecracker, kata container failed.
```
$ docker run --runtime kata-runtime busybox
docker: Error response from daemon: OCI runtime create failed: Failed to 
connect to firecrackerinstance (timeout 10s): unknown.
```
Causes and Fixes are as follows:
1. new flag `-config-file` belongs to the jailed firecracker.
    so, adhering to the `end of command options` convention, we need to give `config-file` a 
prefix `--`.
2. The path of the config file(`fcConfig.json`) should be also relative to the jailed firecracker.
3. Since we do the configuration before func `fcInit` now, we also need to bring `jailer check` ahead.

### CODE REVIEW

File: virtcontainers/fc.go [hunk 1]
```diff
@@ -303,6 +303,7 @@ func (fc *firecracker) newFireClient() *client.Firecracker {
 func (fc *firecracker) vmRunning() bool {
 	resp, err := fc.client().Operations.DescribeInstance(nil)
 	if err != nil {
+		fc.Logger().WithError(err).Error("getting vm status failed")
```
Comment (jodh-intel) [inline virtcontainers/fc.go]: Nice!

**Comment by devimc** [comment]
thanks @Pennyzct , seems like CI is not happy

```
virtcontainers/fc.go:713: File is not `gofmt`-ed with `-s` (gofmt)

        if fc.config.JailerPath != "" {

                fc.jailed = true

        }
```

**Comment by Pennyzct** [comment]
Hi @devimc 
so sorry... my neglect. updated. ;).

**Comment by Pennyzct** [comment]
Hi guys
Maybe we should integration test for running firecracker with jailer, then the bug like this will be detected. ;).

**Comment by Pennyzct** [comment]
### Update:
The config file `fcConfig.json` should be unmounted and cleaned up. ;)
/test-fc

**Comment by Pennyzct** [comment]
Hi~ guys
All green. ;).
But need another reviewer, @kata-containers/runtime. ;)

**Pull request merged**